modules/mcuboot/CMakeLists.txt

```diff
@@ -192,6 +192,12 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
       ${ZEPHYR_NRF_MODULE_DIR}/modules/mcuboot/tfm.conf
       )
   endif()
+  if (CONFIG_SPM)
+    add_overlay_config(
+      mcuboot
+      ${ZEPHYR_NRF_MODULE_DIR}/modules/mcuboot/spm.conf
+      )
+   endif()
```

Comment:
this commit af2f52a105507b2842e5d1e8dc454fffe1a051e2 only mentions partition manager / pcd_sram.

So what is the relation to `spm.conf` and why SPM knowledge this needed in context of MCUBoot ?

Reply:
It mentions trustzone does it not? SPM/TF-M enables you to run trustzone.
same with the modules/mcuboot/tfm.conf  file.

The reason we need to pass this information to the context of MCUBoot is that it's MCUBoot image which adds `pm.yml.pcd` which means the `autoconf.h` file used is the one from MCUBoot. Therefor we need to inform MCUBoot wether we are using Trustzone or not (secure/non-secure). So that `pm.yml.pcd` can place `pcd_sram` in the desired location, which is either at the end of RAM or end of SECURE RAM.

Reply:
> It mentions trustzone does it not?

It does, but the partition manager runs as part of the parent build not the MCUboot build.
Only preprocessing of the `pm.yml.*` files are done here before being shared to parent.

> The reason we need to pass this information to the context of MCUBoot is that it's MCUBoot image which adds pm.yml.pcd which means the autoconf.h file used is the one from MCUBoot.

but why is it needed to share the same file twice ?

and besides the `pm.yml.pcd` then the only place I see `CONFIG_TRUSTED_EXECUTION_SECURE` used in MCUboot, is for sharing of `pm.yml.trustzone` (code shared by all images):
https://github.com/nrfconnect/sdk-nrf/blob/246a91690a1307c598133ee3d64bbc5b0568bc06/subsys/partition_manager/CMakeLists.txt#L88-L90

but the same file is being shared by `spm` (after preprocessing in both cases).
This is easily seen in a CMake trace:
```
# from spm/shared_vars.cmake
/projects/github/ncs/nrf/build/spm/shared_vars.cmake(10):  set_property(TARGET spm_shared_property_target APPEND PROPERTY PM_YML_FILES;/projects/github/ncs/nrf/build/spm/modules/nrf/subsys/partition_manager/pm.yml.trustzone )
...
# from mcuboot/shared_vars.cmake
/projects/github/ncs/nrf/build/mcuboot/shared_vars.cmake(12):  set_property(TARGET mcuboot_shared_property_target APPEND PROPERTY PM_YML_FILES;/projects/github/ncs/nrf/build/mcuboot/modules/nrf/subsys/partition_manager/pm.yml.trustzone )
```

and those two files are identical:
```
$ diff /projects/github/ncs/nrf/build/spm/modules/nrf/subsys/partition_manager/pm.yml.trustzone /projects/github/ncs/nrf/build/mcuboot/modules/nrf/subsys/partition_manager/pm.yml.trustzone
$ 
```

and both are going to the partition_manager script:
```
execute_process(COMMAND /usr/bin/python3.8;/projects/github/ncs/nrf/scripts/partition_manager.py;--input-files;/projects/github/ncs/nrf/build/spm/zephyr/include/generated/pm.yml;/projects/github/ncs/nrf/build/spm/modules/nrf/subsys/partition_manager/pm.yml.trustzone;/projects/github/ncs/nrf/build/mcuboot/zephyr/include/generated/pm.yml;/projects/github/ncs/nrf/build/mcuboot/modules/nrf/subsys/partition_manager/pm.yml.pcd;/projects/github/ncs/nrf/build/mcuboot/modules/nrf/subsys/partition_manager/pm.yml.trustzone;/projects/github/ncs/nrf/build/zephyr/include/generated/pm.yml;/projects/github/ncs/nrf/build/modules/nrf/subsys/partition_manager/pm.yml.settings;/projects/github/ncs/nrf/build/modules/nrf/subsys/partition_manager/pm.yml.rpmsg_nrf53;/projects/github/ncs/nrf/build/modules/nrf/subsys/partition_manager/pm.yml.trustzone;--regions;sram_primary;otp;flash_primary;--output-partitions;/projects/github/ncs/nrf/build/partitions.yml;--output-regions;/projects/github/ncs/nrf/build/regions.yml;--sram_primary-size;0x80000;--sram_primary-base-address;0x20000000;--sram_primary-placement-strategy;complex;--sram_primary-dynamic-partition;sram_primary;--otp-size;764;--otp-base-address;0xff8100;--otp-placement-strategy;start_to_end;--flash_primary-size;0x100000;--flash_primary-base-address;0x0;--flash_primary-placement-strategy;complex;--flash_primary-device;NRF_FLASH_DRV_NAME RESULT_VARIABLE ret )
```

So to solve the need for preprocessing `pm.yml.pcd` based on TF-M, then you suddenly start to preprocess the `pm.yml.trustzone` and share this file twice.

Reply:
Did not see that my bad. I've fixed it now by introducing a new Kconfig to ensure it's not shared twice.

Reply:
This should be fixed now